doc/contributing.rst

```diff
@@ -553,20 +553,40 @@ To keep the tool user friendly, try to avoid making changes that are not
 backward compatible, i.e. changes that require users to change their existing
 recipes, diagnostics, configuration files, or scripts.
 
-If you really must change the public interfaces of the tool, always discuss this
-with the `@ESMValGroup/esmvaltool-coreteam`_.
-Try to deprecate the feature first by issuing a :py:class:`DeprecationWarning`
-using the :py:mod:`warnings` module and schedule it for removal three
-`minor versions <https://semver.org/>`__ from the latest released version.
-For example, when you deprecate a feature in a pull request that will be
-included in version 2.3, that feature could be removed in version 2.5.
+If you really must change the public interfaces of the tool, always discuss
+this with the `@ESMValGroup/esmvaltool-coreteam`_.  Try to deprecate the
+feature first by issuing a
+:class:`esmvalcore.exceptions.ESMValCoreDeprecationWarning` using the
+:py:mod:`warnings` module and schedule it for removal two `minor versions
```

Comment:
```suggestion
:class:`~esmvalcore.exceptions.ESMValCoreDeprecationWarning` using the
:mod:`warnings` module and schedule it for removal two `minor versions
```
Prefixing the fully qualified class name (FQCN) with the `~` will not show the containing packages, thus making it a bit more friendly to read. `:py:` can be omitted because it is [the default domain](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#basic-markup).